QgisModelBaker/libqgsprojectgen/generator/generator.py

```diff
@@ -101,12 +101,19 @@ def layers(self, filter_layer_list=[]):
             if filter_layer_list and record['tablename'] not in filter_layer_list:
                 continue
 
-            alias = record['table_alias'] if 'table_alias' in record else ''
             is_domain = record['kind_settings'] == 'ENUM' or record[
                 'kind_settings'] == 'CATALOGUE' if 'kind_settings' in record else False
             is_structure = record['kind_settings'] == 'STRUCTURE' if 'kind_settings' in record else False
             is_nmrel = record['kind_settings'] == 'ASSOCIATION' if 'kind_settings' in record else False
 
+            alias = record['table_alias'] if 'table_alias' in record else None
+            if not alias:
+                if is_domain:
+                    short_name = record['ili_name'].split('.')[-2] + '_' + record['ili_name'].split('.')[-1] if 'ili_name' in record else ''
```

Comment:
Are we sure they always split into enough parts (and don't suddenly get an array out of bounds exception here?

Reply:
You mean in case the ili_name structure changes? What do you suggest? Make a fallback in case of exception?

Reply:
I'm not absolutely sure we always do have a dot in each name - maybe you know better? - so if we happen to encounter an ili_name `NAME_WITHOUT_DOT` for whatever reason the user will get nothing but an exception which is quite uncool. So unless you are 100% sure, I'd just do the fallback :+1: